src/nvim/cursor_shape.c

```diff
@@ -45,6 +45,7 @@ cursorentry_T shape_table[SHAPE_IDX_COUNT] = {
   { "more", 0, 0, 0,   0,   0,   0, 0, 0, "m", SHAPE_MOUSE },
   { "more_lastline", 0, 0, 0,   0,   0,   0, 0, 0, "ml", SHAPE_MOUSE },
   { "showmatch", 0, 0, 0, 100, 100, 100, 0, 0, "sm", SHAPE_CURSOR },
+  { "terminal", 0, 0, 0, 0, 0, 0, 0, 0, "te", SHAPE_CURSOR },
```

Comment:
Just for reference: interesting that vim doesn't have this https://github.com/vim/vim/blob/c942f84aadffd0c8969ecf81e3e9103722b2714f/src/misc2.c#L2042

Reply:
That's not surprising, we are adding this here only so that we can communicate cursor shape information to the UI in the `mode_info_set` event. Since Vim doesn't have the UI architecture it makes sense they would have another way of implementing this.